File: Day-02-Strings/10-Isomorphic-Strings/Solution.java
import java.util.*;
class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String s1 = sc.next();
        String s2 = sc.next();

        int[] map1 = new int[256];
        int[] map2 = new int[256];

        for(int i = 0; i < s1.length(); i++) {
            char a = s1.charAt(i);
            char b = s2.charAt(i);

            if(map1[a] == 0 && map2[b] == 0) {
                map1[a] = b;
                map2[b] = a;
            } else {
                if(map1[a] != b) {
                    System.out.print("false");
                    return;
                }
            }
        }

        System.out.print("true");
    }
}

File: Day-02-Strings/10-Isomorphic-Strings/Solution.cpp
#include <stdio.h>
#include <string.h>

int main() {
    char s1[10001], s2[10001];
    scanf("%s %s", s1, s2);

    int map1[256] = {0}, map2[256] = {0};

    for(int i = 0; s1[i]; i++) {
        if(map1[s1[i]] == 0 && map2[s2[i]] == 0) {
            map1[s1[i]] = s2[i];
            map2[s2[i]] = s1[i];
        } else {
            if(map1[s1[i]] != s2[i]) {
                printf("false");
                return 0;
            }
        }
    }

    printf("true");
}